flaky test on DDLNotifier

https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_check2/detail/ghpr_check2/27040/pipeline

```
panic: assert failed, error processing events: session pool closed [recovered]

	panic: assert failed, error processing events: session pool closed



goroutine 132297 [running]:

github.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).OnBecomeOwner.func1({0x8beb040, 0xc048a9e3f0})

	pkg/ddl/notifier/subscribe.go:338 +0x45

github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithRecover.func1.1()

	pkg/util/wait_group_wrapper.go:195 +0x43

panic({0x8beb040?, 0xc048a9e3f0?})

	GOROOT/src/runtime/panic.go:785 +0x132

github.com/pingcap/tidb/pkg/util/intest.doPanic(...)

	pkg/util/intest/assert.go:62

github.com/pingcap/tidb/pkg/util/intest.Assert(...)

	pkg/util/intest/assert.go:30

github.com/pingcap/tidb/pkg/ddl/notifier.(*DDLNotifier).start(0xc02fd50120)

	pkg/ddl/notifier/subscribe.go:164 +0x765

github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithRecover.func1()

	pkg/util/wait_group_wrapper.go:199 +0xa6

created by github.com/pingcap/tidb/pkg/util.(*WaitGroupWrapper).RunWithRecover in goroutine 126710

	pkg/util/wait_group_wrapper.go:191 +0x125
```

--- a/pkg/ddl/notifier/subscribe.go
+++ b/pkg/ddl/notifier/subscribe.go
@@ -162,7 +162,9 @@ func (n *DDLNotifier) start() {
 		case <-ticker.C:
 			if err := n.processEvents(ctx); err != nil {
 				intest.Assert(
-					errors.ErrorEqual(err, context.Canceled) || strings.Contains(err.Error(), "mock handleTaskOnce error"),
+					errors.ErrorEqual(err, context.Canceled) ||
+						strings.Contains(err.Error(), "mock handleTaskOnce error") ||
+						strings.Contains(err.Error(), "session pool closed"),
 					fmt.Sprintf("error processing events: %v", err),
 				)
 				logutil.Logger(ctx).Error("Error processing events", zap.Error(err))

--- a/pkg/statistics/handle/ddl/ddl.go
+++ b/pkg/statistics/handle/ddl/ddl.go
@@ -54,7 +54,9 @@ func (h *ddlHandlerImpl) HandleDDLEvent(ctx context.Context, sctx sessionctx.Con
 	err := h.sub.handle(ctx, sctx, s)
 	if err != nil {
 		intest.Assert(
-			errors.ErrorEqual(err, context.Canceled) || strings.Contains(err.Error(), "mock handleTaskOnce error"),
+			errors.ErrorEqual(err, context.Canceled) ||
+				strings.Contains(err.Error(), "mock handleTaskOnce error") ||
+				strings.Contains(err.Error(), "session pool closed"),
 			fmt.Sprintf("handle ddl event failed, err: %v", err),
 		)
 	}

